### DISCUSSION TITLE
Tags

### DISCUSSION THREAD

**Discussion by onordbo** [fremtind/jokul#2341] [Annet]
Som et ledd i [inventory prosjektet](https://github.com/fremtind/jokul/projects/6) vårt ønsker vi tilbakemelding og stemmer fra både design og teknologi for å kartlegge hvilke komponenter og behov vi burde prioritere. Nedenfor følger en kort hypotese/slutning fra Designsystem-teamet, i tillegg til en rekke kommentarer vi har merket oss gjennom arbeidet.

Se på skjermbildene fra inventory [tilgjengelig her for referanse](https://www.figma.com/file/66kOJ5nCHjgtkfLW2sQWB8/Jøkul-Inventory-Innsiktsyntese?node-id=0%3A1). Legg gjerne til ønsker, behov, tanker, meninger og annet som kommenaterer i denne tråden, så vi kan sørge for at oppgavene blir så godt beriket med informasjon som mulig, dette gjør det enklere for alle involverte å ta tak i og løse de utfordringene vi møter sammen.

***

### TL;DR 👉 Denne ser vi flere teams har behov for. Stor visuell spredning p.t

- SAM (ønske/behov): Tags er et behov som er viktig for SAM teamet?
- Motor (ønske/behov): “Tags”, farget statustekst
- Flyt (ønske/behov): Jøkul tags
- Det finnes en sak på dette i GitHub som burde følges opp og prioriteres
- Bør legge føringer for bruk av stor og liten tekst
- Trenger tags en utvidet fargepalett?

Brukes i dag av SAM, SeOpp og Flyt, til en viss grad.

**Behov:** Merkelapp med annen farge enn bakgrunnsfargen. Brukes til korte statustekster som "Påbegynt", "e-kunde", "Ny", "Til utfylling", og lingnende.

Det finnes et tilfelle av ikon brukt inni en tag.

**Comment by wkillerud** [comment]
Den nevnte diskusjonen: https://github.com/fremtind/jokul/issues/1668

**Comment by wkillerud** [comment]
## Forslag til beskrivelse


**Tags er små informasjonsbobler som kan brukes til å presentere en eller flere biter med informasjon tilknyttet et hovedtema.**

Tags inneholder kun tekst. Teksten starter med stor forbokstav etterfulgt av små, med mindre det er en forkortelse eller merkevare. Teksten bør ikke overskride 12 tegn, og mellomrom bør unngås hvis mulig.

Vi har to kategorier av tags:

1.  status - vises med en passende farge
2.  emne, metadata - nøytral

Fargene i statustags har en betydning på samme måte som [Message box](https://jokul.fremtind.no/komponenter/messagebox):

-   Grønt betyr at noe er vellykket eller ferdig
-   Gult informerer om noe som kan ha innvirkning på brukeren, men hvor det ikke kreves en handling
-   Rødt betyr at en handling kreves eller noe har gått galt så handlingen ikke kunne gjennomføres

Blått kan brukes om de andre kategoriene ikke passer, men bare dersom samme sted vil kunne vise en annen statusfarge hvis situasjonen var annerledes. Blått er med andre ord ikke en nøytral farge som kan brukes til informasjon utenfor statuskategorien. Til det brukes emnevarianten.

## Når bruker vi tags?

Før du bruker en tag bør du stille noen kontrollspørsmål:

-   Er teksten kort nok til å passe inn i en tag?
-   Er det tydelig for brukeren hva tagsene er tilknyttet? Også for brukere av skjermlesere?
-   Passer informasjonen jeg skal vise inn i én av de to kategoriene – status og emne?
-   Vises informasjonen i en tabell? I så fall, gir det verdi å bruke en tag i stedet for en vanlig tekstkolonne?

**Reply by onordbo** [reply]
Legger ved et forslag til visuell implementering med basalbehovene dekket; nøytral og et par fargekoder—inspirert av MessageBox komponenten vår. Hva synes dere @Murstam, @Stormoen og @AudunBerdal?

![image](https://user-images.githubusercontent.com/4263132/138860682-8a1f2485-8142-461d-b929-0425da21894f.png)

**Comment by HavardPede** [comment]
Kan kanskje bli litt vanskelig å se forskjell på hva som er Button og hva som er Tag, spesielt ved invertert nøytral.

![image](https://user-images.githubusercontent.com/30959515/138883775-ffa2c6ae-aab7-49c5-a3f2-baaff2d869b6.png)
![image](https://user-images.githubusercontent.com/30959515/138883815-63a27d35-534e-45d6-8b40-d7f9a7b405bf.png)

edit: spesifiserte hvilken type.

**Comment by karimarkhus** [comment]
syns de var kjempefine!!! absolutt et behov for tags! men enig i at de ligner litt på button. Men det kommer kanskje an på størrelsen også?

**Reply by HavardPede** [reply]
Det er et godt poeng. 
Her er et eksempel vi har i SAM. Dette er i en utviklingsbranch, så skal ryddes opp før det er i prod. 

Men her har vi: 
1. Tags noe lignende det dere foreslår, for å beskrive kundeinformasjon (NSF, ekunde toppen av siden).
2. En ny annen versjon av en tag vi testet ut for å skille fra button (Utkast). 
3. Button (Sett til Digital Aksept).
![image](https://user-images.githubusercontent.com/30959515/138886515-acbe1f6a-2731-4b9d-a2b9-0b05aabf9b98.png)

Så det at dere implementerer en Tag er supert, så kan vi erstatte våre. 
Grunnen for at vi implementerte en firkantet tag var for å lettere skille mellom button og tag. 
Er enig @karimarkhus at størrelse vil påvirke dette, samt hvor det er brukt.

**Comment by Stormoen** [comment]
Supert utkast. Jeg er enig i at de blir litt for lik buttons hva gjelder form. Også litt usikker på når man skal bruke "blå info" kontra "nøytral", greit å definere noen regler her så ikke meldingsfargene løper løpsk avgårde :)

**Reply by wkillerud** [reply]
Godt innspill. Jobber med noen "dos and donts" for å prøve å tydeliggjøre hva jeg tenker er forskjellen på blå og nøytral. Ikke sikkert det gir mening da men 😄 

![image](https://user-images.githubusercontent.com/1223410/138889890-49fdd7cc-501f-4e38-8ea8-114fbc04ccbe.png)

**Reply by karimarkhus** [reply]
jeg lurer også på om det går an å ha litt ulike versjoner? f.eks noen ganger har man behov for å vise ulike kategorier med f.eks tags, men kanskje det ikke har noe å gjøre med info, feil, suksess og advarsel

**Reply by Stormoen** [reply]
Veldig gode eksempler alle tre! Det første jeg tenkte på når jeg så farger i tags var at de sikkert blir misbrukt slik som eksempelet "miljøvennlig". Så lenge fargene brukes for å kommunisere på samme måte som meldingsboksene er det innafor tenker jeg.

**Reply by wkillerud** [reply]
Kategorier tenkte vel @onordbo og jeg faller under emner, så nøytral farge. Trengs det farger for å differensiere? Åpen for å overbevises, vi ønsket bare å starte med så få varianter som mulig 😄 @karimarkhus

**Reply by karimarkhus** [reply]
@wkillerud tanken var bare at hvis man f.eks skal merke et kort med flere "tags" så kunne det vært fint å ha ulike kategorier sånn at man lettere kan "skanne" innholdet.

**Reply by onordbo** [reply]
Dette er et veldig godt innspill vi burde ta med—la oss se på noen variasjoner sammen i neste iterasjon @karimarkhus 🎯

**Comment by Stakset** [comment]
Litt skeptisk til at de har samme avrunding som knappene. Prøvde det i noen eksempler og føler det blir en konflikt med knapp og toggle switch. Hva om de heller hadde samme avrunding som kortene våre? 

En annen idé kunne vært å ha nøytral i fargen varde slik at man kan bruke sort skrift på alle, og likt mellom light og dark mode. 

![Tags](https://user-images.githubusercontent.com/72978199/138889058-83caf962-b610-4eec-8745-48f3bbee0250.png)

**Reply by Stormoen** [reply]
Liker at de tar igjen formen fra meldingsboksene. Og enig i varde, slik at differansen til knapp også blir større.

**Reply by onordbo** [reply]
+1 liker definitivt bedre litt mindre avrunding, godt innspill @Stakset

**Comment by onordbo** [comment]
Oppdatert! [https://www.figma.com/file/eL15GtAzhsUk9MHWxZ92DP/tags?node-id=1997%3A6990](https://www.figma.com/file/eL15GtAzhsUk9MHWxZ92DP/tags?node-id=1997%3A6990)

cc @Stakset @Stormoen

**Reply by onordbo** [reply]
![image](https://user-images.githubusercontent.com/4263132/138893921-6ea44dda-c62e-4061-a3c8-635d2857e567.png)

**Reply by Stakset** [reply]
Liker tempoet deres :) Godt jobba!

**Reply by HavardPede** [reply]
👏

**Reply by piofinn** [reply]
Smud! Hva med å teste den i UU-guiden i Portalen? Enkel måte å få testet dark mode på også.

![Skjermbilde 2021-10-26 kl  16 52 14](https://user-images.githubusercontent.com/25739615/138904780-b071adc8-4a76-45b3-9bee-b040f3050918.png)

Ikke helt tilfeldig valg av tema i screenshot-et heller 😆

**Reply by onordbo** [reply]
Ser ut som vi er innafor på kontrast hvertfall, @piofinn 😉 

![Kapture 2021-10-26 at 17 16 43](https://user-images.githubusercontent.com/4263132/138909876-0b8ee39f-53d7-4fac-a449-96d5e06ce6c9.gif)

**Comment by Murstam** [comment]
Bra jobba! Enig i form og farge :) Tenker teksten kan være i Granitt for det er vel all tekst ellers nå. 

Testet den litt i tabell og førsteinntrykket mitt er at den er litt stor, spesielt på compact? Et alternativ kan være å kutte padding fra 12/4 til 8/2. Også kan det hende at teksten holder i regular, men litt usikker på den. 

Legger ved nåværende pluss to alternativer
<img width="966" alt="Nå" src="https://user-images.githubusercontent.com/47025023/138957461-5e9ce4c2-7003-4b0e-b96e-156945a934ab.png">
<img width="990" alt="Alt 1" src="https://user-images.githubusercontent.com/47025023/138957470-15bb6fde-0762-45f4-8543-1bcd5504637f.png">
<img width="1017" alt="Alt 2" src="https://user-images.githubusercontent.com/47025023/138957473-9659b420-dc30-43e3-adc7-d9f53be6165b.png">
.

**Reply by wkillerud** [reply]
Vi tenkte bold for lesbarheten på farget bakgrunn, og spesielt til hvit-på-svart.

Enig i at paddingen kan virke stor i kompakttabell, men siterer meg sjæl og stiller et spørsmål 😉 @Murstam 

> Vises informasjonen i en tabell? I så fall, gir det verdi å bruke en tag i stedet for en vanlig tekstkolonne?

Lov å svare "ja og ja" 😄

Det sagt syntes jeg 8/2 padding kan funke, også med bold tekst om vi ønsker å beholde det.

**Reply by onordbo** [reply]
Enig at den kan virke litt stor i noen kontekster, la oss se litt på om vi kan skru til paddingen enda mer. Dog tror jeg definitivt vi burde holde på bold for lesbarhetens skyld, spesielt om vi gjør tag'en enda mer kompakt.

**Reply by Stakset** [reply]
Må være granitt ja. 

Kunne vi hatt varianter av tags? Slik vi gjør message box. Compact kunne vært 8/2, også kunne vi hatt mulighet til å ha regular. Den stjeler veldig mye fokus med bold slik det er satt i tabellen nå ihvertfall. Føler ikke det har noe å si for det overordna uttrykket at vi har varianter. Så lenge folk ikke blander, men det gjelder alle komponenter :)

**Reply by piofinn** [reply]
> kunne vi hatt mulighet til å ha regular

Jeg legger merke til at alle eksemplene på nåværende implementasjoner i denne tråden er med normal vekt. Kanskje vi kunne fått sett noen eksempler på de nye i dark mode med normal fontvekt @onordbo, siden det virker som det var der det smertet mest? (Skjønner at selve komponenten er lik, men kontrasten kan oppfattes annerledes mot mørk bakgrunn).

**Reply by Stakset** [reply]
Ganske slitsomt å lese regular 16 px tags i darkmode. Det gjelder egentlig for dagens accessibility-side i darkmode også. 
 ![Darkmode](https://user-images.githubusercontent.com/72978199/139038610-a4ae209f-8de9-435f-bc4b-882de7e546cd.png)
![Skjermdump](https://user-images.githubusercontent.com/72978199/139039525-a96f86e2-392b-49f1-85a1-d31e266636ad.png)

**Reply by Stormoen** [reply]
Fint og enig med kompaktversjonen av tags! Synes kanskje det holder med bold tekst. Hvis man velger å bruke tags så tenker jeg at meninga er at det er skal fremheves. Nå har vi jo samme versjon for mørk og lys bakgrunn, så da er det blir det vanskelig å forhindre "feilbruk" med regular tekst på mørk bakgrunn.
